org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/panel/InfoPanel.java

```diff
@@ -2991,7 +2995,16 @@ public void sort(Comparator<Object> cmpr, boolean ascending) {
 		int col = lsc.getColumnIndex();
 		indexOrderColumn = col;
 		isColumnSortAscending = ascending;
-		sqlOrderColumn = p_layout[col].getColSQL().trim();
+		String displayColumn = p_layout[col].getDisplayColumn();
+		sqlOrderColumn = !Util.isEmpty(displayColumn) ? displayColumn : p_layout[col].getColSQL().trim();
+
+		String columnName = p_layout[col].getColumnName();
+		if(columnName.endsWith("_ID")) {
+			int index = columnName.lastIndexOf("_ID");
+			String tableName = columnName = columnName.substring(0, index);
```

Comment:
This doesn't always work, for e.g, you have AD_Reference_Value_ID but the foreign table is AD_Reference, not AD_Reference_Value.

Reply:
I added getByKeyColumns(keyColumns) method to MTable to fix that

Reply:
You have ad_reference_id and ad_reference_value_id in ad_infocolumn, should get the table name from there instead. Also, with the presence of ad_reference_id, we should check against that instead of using the "_ID" pattern.

Reply:
Do you mean getting the table name from AD_Ref_Table? I've tested, and when on Info Column I have Table Direct as AD_Reference_ID, then AD_Reference_Value_ID is 0, so I can't get the table name from AD_Ref_Table in this case.

Reply:
If ad_reference_value_id is 0, you get the table name from column name. That must work because otherwise there's no other place to get the table name from.

Reply:
Ok thanks, done